### PR TITLE
Update steps to build Chapel with newer HDFS sources

### PR DESCRIPTION
**Pull request by majetideepak**
The current instructions to build HDFS seem to be outdated. This patch updates the process.

### CODE REVIEW

**Comment by bradcray** [comment]
@ct-clmsn : Would you be able to review this patch?

File: modules/packages/HDFS.chpl [hunk 1]
```diff
@@ -49,6 +49,10 @@ CHPL_AUX_FILESYS to 'hdfs' to enable HDFS support:
 
   export CHPL_AUX_FILESYS=hdfs
 
+The HDFS C API on some platforms freezes using the default q-threads. Use fifo instead.
```
Comment (mppf) [inline modules/packages/HDFS.chpl]: q-threads -> qthreads

**Comment by mppf** [comment]
@majetideepak or @ct-clmsn, it seems to me that the documentation should require a specific version of HDFS / libhdfs3. Also, @majetideepak, what testing did you do with these changes? Which versions of HDFS and libhdfs3 work?

**Comment by majetideepak** [comment]
I have been wanting to reply to this after testing with libhdfs3, but did not get around to finishing it.
I tested this on apache hadoop 2.7.2, which provides libhdfs (C Wrapper over Java). I wasn't aware the original documentation was for libhdfs3. I will try to finish testing libhdfs3 soon and update this patch with instructions for both (libhdfs and libhdfs3).

**Comment by mppf** [comment]
hi @majetideepak - sounds good, thanks!

**Comment by mppf** [comment]
hi @majetideepak - any progress on this PR? We're preparing for a release and it would be nice to have current hdfs/libhdfs support. Thanks!

**Comment by majetideepak** [comment]
@mppf Apologies for the delay. I will finish this patch by the end of this week. Will that work ?

**Comment by mppf** [comment]
@majetideepak - yes, sounds great, thanks very much.

File: modules/packages/HDFS.chpl [hunk 2]
```diff
@@ -176,7 +182,7 @@ First reflect your directory structure and version numbers (etc) in the
 
 2. Install Hadoop
```
Comment (majetideepak) [inline modules/packages/HDFS.chpl]: This file does not have anything about libhdfs3. According to http://chapel.cray.com/docs/1.13/technotes/auxIO.html#parallel-and-distributed-i-o-features, the file `$CHPL_HOME/third-party/libhdfs3/README` describes the instructions for libhdfs3.

**Comment by majetideepak** [comment]
@mppf I tested read/write to hdfs using libhdfs3. The instructions for libhdfs3 located at `$CHPL_HOME/third-party/libhdfs3/README` are correct.

**Comment by mppf** [comment]
@majetideepak - thanks very much! I'm asking @ben-albrecht to have a final look at this PR before merging, but it looks good to me.

File: modules/packages/HDFS.chpl (hunk 2 above)
Comment (ben-albrecht) [inline modules/packages/HDFS.chpl]: Fixed in #4501

**Comment by ben-albrecht** [comment]
Looks good to me. Thanks for the contribution @majetideepak

**Pull request merged**